components/camel-jooq/src/main/java/org/apache/camel/component/jooq/JooqConsumer.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.component.jooq;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.Processor;
+import org.apache.camel.support.ScheduledPollConsumer;
+
+public class JooqConsumer extends ScheduledPollConsumer {
+    private final Processor processor;
+    private final JooqEndpoint jooqEndpoint;
+
+    public JooqConsumer(JooqEndpoint jooqEndpoint, Processor processor) {
+        super(jooqEndpoint, processor);
+        this.jooqEndpoint = jooqEndpoint;
+        this.processor = processor;
+    }
+
+    @Override
+    public Processor getProcessor() {
+        return processor;
+    }
+
+    @Override
+    public Endpoint getEndpoint() {
+        return jooqEndpoint;
+    }
+
+    @Override
+    protected int poll() {
+        return 0;
```

Comment:
Why this consumer always return 0? If the consumer doesn't do anything, it's better to avoid its support.

Reply:
you can raise an UnsupportedOperationException in your createConsumer.

`throw new UnsupportedOperationException("You cannot receive messages from this endpoint");`

Reply:
Yes, consumer doesent do anything. I'll delete it. I'll implement it later with task CAMEL-13261.